src/test/java/ch/uzh/ifi/hase/soprafs22/service/UserServiceTest.java

```diff
@@ -165,9 +263,413 @@ void updateUser_Conflict(){ //Test what happens when user tries to change userna
+        assertEquals(testUser , isMatch.getUserPair().getObj1());
+        assertEquals(otherUser, isMatch.getUserPair().getObj2());
+    }
+
+    @Test
+    void deleteGameIfEmpty_emptyGame(){
+      Game emptyGame = new Game();
+      testUser.setActiveGame(emptyGame);
+      //This moves the emptyGame to past Games
+      testUser.flushGameToPastGames();
+      userService.deleteGameIfEmpty(testUser, emptyGame);
+      //getPastGames should now be empty as the emptygame got deleted
+      assertEquals(Collections.emptyList(), testUser.getPastGames());
+    }
+
+    @Test
+    void deleteGameIfEmpty_nonEmptyGame(){
+        Play play = new Play();
+        Game nonEmptyGame = new Game();
+        nonEmptyGame.enqueuePlay(play);
+        testUser.setActiveGame(nonEmptyGame);
+        //This moves the emptyGame to past Games
+        testUser.flushGameToPastGames();
+        userService.deleteGameIfEmpty(testUser, nonEmptyGame);
+        //Should now not return empty list for past games
+        assertNotEquals(Collections.emptyList(), testUser.getPastGames());
+    }
+
+    @Test
+    void updateActiveGameIfNecessary_oldGame() {
+      testGame.setCreationTime(new Date(0,0,0));
+      testUser.setActiveGame(testGame);
+      long id = testUser.getId();
+
+      Mockito.when(userRepository.findById(id)).thenReturn(testUser);
 
-        assertThrows(ResponseStatusException.class, () -> userService.updateUser(putUser));
+      userService.updateActiveGameIfNecessary(testUser.getId());
+        assertNull(testUser.getActiveGame());
     }
 
+    @Test
+    void updateActiveGameIfNecessary_newGame() {
+        testUser.setActiveGame(testGame);
+
+        long id = testUser.getId();
+
+        Mockito.when(userRepository.findById(id)).thenReturn(testUser);
+
+        userService.updateActiveGameIfNecessary(testUser.getId());
+        assertEquals(testGame, testUser.getActiveGame());
+    }
+
+    @Test
+    void setMatch(){
+      Match testMatch =  userService.createMatch(testUser, otherUser);
+      testMatch.setMatchId(2L);
+
+      //both users should now have the testmatch
+      userService.setMatch(testMatch);
+
+      //checking if both users have the match with the given id
+      assertTrue(testUser.getMatches().contains(testMatch.getMatchId()));
+        assertTrue(otherUser.getMatches().contains(testMatch.getMatchId()));
+    }
+
+    @Test
+    void otherUserLikesUser_true(){
+      //set the otheruser to like testUser
+      userService.setUserLikesOtherUser(otherUser, testUser);
+      assertTrue(userService.otherUserLikesUser(testUser, otherUser));
+    }
+
+    @Test
+    void otherUserLikesUser_false(){
+        //testUser by default isn't liked by other user
+        assertFalse(userService.otherUserLikesUser(testUser, otherUser));
+    }
+
+    @Test
+    void isGameBelongingToUSer_true(){
+      //by default we set testgame to belong to user
+        testGame.setUserId(1L);
```

Comment:
instead of 1L, take testUser.getUserId here

Reply:
see new commit